### PR TITLE
memset_sanity adding bzero

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Comment by jvoisin** [comment]
I don't think it should be added:

> The bzero() function is deprecated (marked as LEGACY in POSIX.1-2001); use [memset(3)](https://man7.org/linux/man-pages/man3/memset.3.html) in new programs.  POSIX.1-2008 removes the specification of bzero().  The bzero() function first appeared in 4.3BSD.

**Comment by devnexen** [comment]
I know about this but it s more for wrapping it in existing code, freebsd still uses it intensively. however maybe not so much linux.

**Comment by jvoisin** [comment]
[FreeBSD's manpage]( https://man.freebsd.org/cgi/man.cgi?query=bzero&sektion=3&manpath=FreeBSD+13.2-RELEASE ) also recommends `memset` instead when portability is desired.

If we want to move forward with this, I'd suggest putting it behind an `ifdef freebsd`

**Comment by jvoisin** [comment]
Maybe add a comment like `bzero is removed from the POSIX standard in IEEE Std 1003.1-2008, but still a valid *BSD extension`, and add something like `#if (__FreeBSD__ || __NetBSD__ || __DragonFly__ || __OpenBSD__)`.

File: tests/memset_sanity.c [hunk 1]
```diff
@@ -17,6 +17,9 @@ int main(int argc, char *argv[]) {
     }
 
     p = (uint8_t *) iso_alloc(32);
+#if(__FreeBSD__ || __NetBSD__ || __OpenBSD__ || __DragonFly__)
+    bzero(p, 32);
```
Comment (struct) [inline tests/memset_sanity.c:21]: Why not create a new test just for `bzero` ?

Comment (devnexen) [reply tests/memset_sanity.c:21]: it was for simplicity sake :-)

**Comment by struct** [comment]
I didn't notice in https://github.com/struct/isoalloc/actions/runs/5134126480/jobs/9238551029 but we broke musl builds by looking for `linux/version.h`. We can merge this as-is and fix that in another PR.

**Pull request merged**